src/net_processing.cpp

```diff
@@ -1975,6 +1977,51 @@ static bool ProcessGetCFilters(CNode* pfrom, CDataStream& vRecv, const CChainPar
     return true;
 }
 
+static bool ProcessGetCFHeaders(CNode* pfrom, CDataStream& vRecv, const CChainParams& chain_params,
+                                CConnman* connman)
+{
+    uint8_t filter_type_ser;
+    uint32_t start_height;
+    uint256 stop_hash;
+
+    vRecv >> filter_type_ser >> start_height >> stop_hash;
+
+    BlockFilterType filter_type = static_cast<BlockFilterType>(filter_type_ser);
+
+    const CBlockIndex* stop_index;
+    BlockFilterIndex* filter_index;
+    if (!PrepareBlockFilterRequest(pfrom, chain_params, filter_type, start_height, stop_hash,
+                                   MAX_GETCFHEADERS_SIZE, stop_index, filter_index)) {
+        // Return true because the issue with the invalid request has already been logged.
+        return true;
+    }
+
+    uint256 prev_header;
+    if (start_height > 0) {
+        const CBlockIndex* prev_block = stop_index->GetAncestor(start_height - 1);
+        if (!filter_index->LookupFilterHeader(prev_block, prev_header)) {
+            return error("Failed to find block filter header in index: filter_type=%s, block_hash=%s",
+                         BlockFilterTypeName(filter_type), prev_block->GetBlockHash().ToString());
+        }
+    }
+
+    std::vector<uint256> filter_hashes;
```

Comment:
07dd762

nit: filter_headers, also log message

Reply:
No, I think `filter_hashes` is right.